components/UserContext.js

```diff
@@ -0,0 +1,56 @@
+/* eslint-disable react/destructuring-assignment */
+/* eslint-disable react/no-unused-state */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import Router from 'next/router';
+import Cookie from 'js-cookie';
+
+const UserContext = React.createContext({});
+
+class UserProvider extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      username: {},
+      loggedIn: false
+    };
+
+    this.logIn = (data) => {
+      this.setState({
+        username: { ...data },
+        loggedIn: true
+      }, () => {
+        Cookie.set('user', data);
+        Router.push('/');
```

Comment:
I don't like this part too much, because what if you are not on the Homepage when you login... you will be taken to the homepage, but whatever for now